lib/core/src/Cardano/Wallet/Api/Types.hs

```diff
@@ -195,6 +200,31 @@ import qualified Data.ByteString.Lazy as BL
 import qualified Data.Text as T
 import qualified Data.Text.Encoding as T
 
+{-------------------------------------------------------------------------------
+                               Style of Wallets
+-------------------------------------------------------------------------------}
+
+data WalletStyle
+    = Shelley
+    | Random
+    | Icarus
+    | Trezor
+    | Ledger
+
+type family WalletStyleSymbol (style :: WalletStyle) :: Symbol where
+    WalletStyleSymbol 'Shelley = "shelley"
+    WalletStyleSymbol 'Random  = "random"
+    WalletStyleSymbol 'Icarus  = "icarus"
+    WalletStyleSymbol 'Trezor  = "trezor"
+    WalletStyleSymbol 'Ledger  = "ledger"
```

Comment:
I wonder, is there a way to get these conversions "for free", without having to define them manually?

Reply:
There are ways. Though we are at the type-level here, so things get a little more complicated. I found that simple type-family to be a nice middleground :)

Reply:
I am also here for explicite saying what type level string looks like. Introducing type level function doing the same probably would only decrease readibility of this part of code.